src/modalities/util.py

```diff
@@ -76,6 +76,12 @@ def get_total_number_of_trainable_parameters(model: FSDP) -> Number:
     num_params_tensor = torch.tensor(num_params).cuda()
     dist.all_reduce(num_params_tensor, op=dist.ReduceOp.SUM)
     total_num_params = num_params_tensor.item()
+    # For HYBRID sharding, divide by sharding factor to get the correct number of parameters
+    if model.sharding_strategy.name == "HYBRID_SHARD":
+        # Assumes that CUDA is available and each node has the same number of GPUs
+        sharding_factor_hybrid_sharding = torch.distributed.get_world_size() // torch.cuda.device_count()
```

Comment:
```suggestion
        sharding_factor_hybrid_sharding = dist.get_world_size() // torch.cuda.device_count()
```
`torch.distributed` is imported and used as `dist` a few times already in this module

Reply:
Fixed in a04468762f870022806c67f219083ca6068f2cac.